server/models/validations/notification.js

```diff
@@ -0,0 +1,32 @@
+const __ = require('config').universalPath
+const _ = __.require('builders', 'utils')
+const { pass, passFromFunction } = require('./common')
+const { types } = require('../attributes/notification')
+const Group = require('../group')
+
+const validations = module.exports = {
+  pass,
+  type: type => types.includes(type),
+  data: (data, { type }) => {
+    passFromFunction('data', data, _.isPlainObject)
+    dataValidationPerType[type](data)
+    return true
+  }
+}
+
+const dataValidationPerType = {
+  friendAcceptedRequest: ({ user }) => {
+    validations.pass('userId', user)
+  },
+  groupUpdate: ({ group, user, attribute, previousValue, newValue }) => {
+    validations.pass('groupId', group)
+    validations.pass('userId', user)
+    passFromFunction('attribute', attribute, _.isString)
```

Comment:
that's new. is there a reason to not have a simple `  string: _.isString,` in `validations/common.js` ?

Reply:
I'm not exactly sure now, but I think the main motive was to experiment with more direct way to validate, that would not rely on this `this` witchcraft, giving more flexibility if the time comes were an other notification type uses the `attribute` field differently... but yeah `attribute` will always be a string so maybe that's not the best example

Reply:
true, it was a syntax question but youre right `attribute` is completely controlled by the system, there is no exterior input. I would delete this line.

Reply:
also, as a purely opinionated comment : meta programming such as this confuses newbies by making is harder to read (and i say that coming from the ruby world...)

Reply:
I meant `attribute` in any model will be of type string, so it can be added to `validations/common.js`, but I'm in favor of keeping this line as it guards against undefined/non-string values passing through (same thinking as expressed in https://github.com/inventaire/inventaire/pull/480#discussion_r564382134)

Reply:
`passFromFunction` was removed in 6968834